### ISSUE TITLE
Is there a cop that recommends native `String#delete` over `String#remove` from rails?

### ISSUE THREAD

**Issue by PikachuEXE** [rubocop-hq/rubocop#5688]
This is either a `question` or a `feature-request`

The cop I'm inquiring about will flag the following code and provide recommendations accordingly
```ruby
string.remove(char)
# => Use `delete` instead of `remove`
```
This is due to `remove` uses `gsub!`
Using `remove` with strings **only** is slower then using `delete`

I understand that there is [Performance/StringReplacement]`https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/performance/string_replacement.rb`
But it only covers `gsub` and `String#remove` is rails only

**Comment by rrosenblum** [comment]
There is not a cop for this. The functionality of `remove` actually allows it to take in multiple arguments and it calls `gsub!` using each of them. This functionality could be added to `Performance/StringReplacement` with a configuration option or a Rails specific cop could be created to handle the functionality. 

I have some concerns that this may produce some false positives because `remove` seems like it could be generic enough to be defined by other classes as well.

https://github.com/rails/rails/blob/63ec63ca8c1aa8eb222bba51ba4d66a0edf1d585/activesupport/lib/active_support/core_ext/string/filters.rb#L32

**Comment by bbatsov** [comment]
I'm closing this ticket due to no recent activity. Feel free to re-open it if you ever come back to it.
